test/controllers/api-key.test.js

```diff
@@ -24,60 +34,167 @@ describe('ApiKeyController tests', () => {
+      requestContext.pathInfo.headers['x-ims-gw-org-id'] = '';
+      const response = await apiKeyController.createApiKey({ ...requestContext });
+      expect(response.status).to.equal(STATUS_UNAUTHORIZED);
+    });
+
+    it('should throw an error if the organization is not found', async () => {
+      context.imsClient.getImsUserProfile.returns({ organizations: [] });
+      const response = await apiKeyController.createApiKey({ ...requestContext });
+      expect(response.status).to.equal(STATUS_UNAUTHORIZED);
+    });
+
+    it('should throw an error if bearer token is missing', async () => {
+      requestContext.pathInfo.headers.authorization = '';
+      const response = await apiKeyController.createApiKey({ ...requestContext });
+      expect(response.status).to.equal(STATUS_UNAUTHORIZED);
+    });
+
+    it('should create a new API key', async () => {
+      context.dataAccess.getApiKeysByImsUserIdAndImsOrgId.returns([]);
+      const response = await apiKeyController.createApiKey({ ...requestContext });
+      expect(response.status).to.equal(STATUS_CREATED);
+    });
+
+    it('should throw an error if the number of domains exceeds the limit', async () => {
+      requestContext.data.domains = ['https://example.com', 'https://another.com'];
+      const response = await apiKeyController.createApiKey({ ...requestContext });
+      expect(response.status).to.equal(STATUS_FORBIDDEN);
+    });
+
+    it('should throw an error if the user has reached the maximum number of API keys', async () => {
+      context.dataAccess.getApiKeysByImsUserIdAndImsOrgId
+        .returns([createApiKey(exampleApiKey),
+          createApiKey(exampleApiKey), createApiKey(exampleApiKey)]);
+      const response = await apiKeyController.createApiKey({ ...requestContext });
+      expect(response.status).to.equal(STATUS_FORBIDDEN);
     });
   });
 
   describe('deleteApiKey', () => {
-    it('should throw a not implemented error', async () => {
+    it('should delete an API key', async () => {
+      context.dataAccess.getApiKeyById.returns({
+        getImsUserId: () => 'test@example.com',
+        getImsOrgId: () => 'test-org',
+        updateDeletedAt: sinon.stub(),
+      });
+
       const response = await apiKeyController.deleteApiKey({ ...requestContext });
-      expect(response.status).to.equal(501);
+      expect(response.status).to.equal(STATUS_NO_CONTENT);
     });
 
-    it('should throw a not implemented error when id is null', async () => {
-      requestContext.params.id = null;
+    it('should throw an error if the API key is not found', async () => {
+      context.dataAccess.getApiKeyById.returns({
+        getImsUserId: () => 'other@example.com',
+        getImsOrgId: () => 'other-org',
+      });
+
       const response = await apiKeyController.deleteApiKey({ ...requestContext });
-      expect(response.status).to.equal(501);
+      expect(response.status).to.equal(STATUS_FORBIDDEN);
```

Comment:
404, I think would be more correct here.

Reply:
Done.